SimPEG/Mesh/DiffOperators.py

```diff
@@ -565,7 +565,58 @@ def projNeumannOut(n, bc):
 
         return Pbc, Pin, Pout
 
-
+    
+    def unitCellGradx():
+        doc = """Cell centered Gradient in the x dimension used for 
+                regularization. The gradient operator is square (nC-by-nC)"""
+        def fget(self):
+            if self.dim < 3: return None
```

Comment:
Yea correct, needs to be generalized to 1D and 2D.
I can have a look at it later.